ansible_base/authentication/utils/authentication.py

```diff
@@ -95,13 +95,14 @@ def check_system_username(uid: str) -> None:
 
 
 def determine_username_from_uid_social(**kwargs) -> dict:
-    uid_field = getattr(kwargs.get('backend', None), 'ID_KEY', 'username')
-    if uid_field is None:
-        uid_field = 'username'
+    backend = kwargs.get('backend', None)
+    uid_field = 'username'
+    if backend and hasattr(backend, "setting"):
+        uid_field = backend.setting('ID_KEY', default='username')
```

Comment:
Is there a case where `ID_KEY` might be directly in backend that we need to cover as well?

Reply:
For example: https://github.com/python-social-auth/social-core/blob/master/social_core/backends/battlenet.py

Reply:
You are correct. Updating this now with more faviorable logic. This one setting fetch seems to have a ton of cases to consider.